job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy

```diff
@@ -21,69 +36,69 @@ pipeline{
     // IMPORTANT: In case you trigger a new branch here, please create the same branch on build-configuration project
     
     stages {
-        stage('trigger nightly job main') {
+        stage('trigger nightly job ${NEXT_PRODUCT_BRANCH}') {
             steps {
-                build job: 'nightly/main', propagate: false, wait: true, parameters: [
-                        [\$class: 'StringParameterValue', name: 'KIE_GROUP_DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-rhba-main/content-compressed'],
-                        [\$class: 'StringParameterValue', name: 'UMB_VERSION', value: 'main'],
-                        [\$class: 'StringParameterValue', name: 'PRODUCT_VERSION', value: "${PRODUCT_VERSION}"],
+                build job: 'nightly/${NEXT_PRODUCT_BRANCH}', propagate: false, wait: true, parameters: [
+                        [\$class: 'StringParameterValue', name: 'KIE_GROUP_DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-rhba-${NEXT_PRODUCT_BRANCH}/content-compressed'],
+                        [\$class: 'StringParameterValue', name: 'UMB_VERSION', value: '${NEXT_PRODUCT_BRANCH}'],
+                        [\$class: 'StringParameterValue', name: 'PRODUCT_VERSION', value: "${NEXT_PRODUCT_VERSION}"],
                         [\$class: 'StringParameterValue', name: 'DEFAULT_CONFIG_BRANCH', value: "\${env.DEFAULT_CONFIG_BRANCH}"],
                         [\$class: 'BooleanParameterValue', name: 'SKIP_TESTS', value: true]
                 ]
             }
         }
 
         // Kogito prod nightlies
-        stage('trigger kogito nightly job main') {
+        stage('trigger kogito nightly job ${KOGITO_NEXT_PRODUCT_BRANCH}') {
             steps {
-                build job: 'kogito.nightly/main', propagate: false, wait: true, parameters: [
-                        [\$class: 'StringParameterValue', name: 'RHBA_MAVEN_REPO_URL', value: 'http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/rhba-main-nightly-with-upstream'],
-                        [\$class: 'StringParameterValue', name: 'RHBA_VERSION_PREFIX', value: "${RHBA_VERSION_PREFIX}"],
-                        [\$class: 'StringParameterValue', name: 'KOGITO_DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-kogito-main/content-compressed'],
-                        [\$class: 'StringParameterValue', name: 'UMB_VERSION', value: 'main'],
-                        [\$class: 'StringParameterValue', name: 'PRODUCT_VERSION', value: "${PRODUCT_VERSION}"],
-                        [\$class: 'StringParameterValue', name: 'OPTAPLANNER_PRODUCT_VERSION', value: "${PRODUCT_VERSION}"],
+                build job: 'kogito.nightly/${KOGITO_NEXT_PRODUCT_BRANCH}', propagate: false, wait: true, parameters: [
```

Comment:
Should the references to variables be in double quotes or are single quites fine? Sorry, not sure about pipelines..

Reply:
I know it's hard to get it sometimes @winklerm but the whole jenkins pipeline content is part of the `metaJob` variable which uses triple double quoted `"""` stuff https://groovy-lang.org/syntax.html#_triple_double_quoted_string so variables are going to be replaced during prod_cron_meta_nightly_pipeline.groovy execution.
it is different from `\${env.DEFAULT_CONFIG_BRANCH}` which is going to be executed at runtime.

Reply:
I see, thanks!